File: lib/presentation/widgets/app_banner.dart
import 'package:flutter/material.dart';

enum BannerType { info, success, warning, error }

/// A generic inline status banner (info / success / warning / error)
/// with an optional dismiss button.
class AppBanner extends StatelessWidget {
  const AppBanner({
    super.key,
    required this.message,
    this.type = BannerType.info,
    this.onClose,
  });

  final String message;
  final BannerType type;
  final VoidCallback? onClose;

  ({Color bg, Color fg, IconData icon}) _style() {
    switch (type) {
      case BannerType.success:
        return (bg: const Color(0xFFE6F4EA), fg: const Color(0xFF1E7E34), icon: Icons.check_circle);
      case BannerType.warning:
        return (bg: const Color(0xFFFFF4E5), fg: const Color(0xFFB26A00), icon: Icons.warning_amber_rounded);
      case BannerType.error:
        return (bg: const Color(0xFFFDECEA), fg: const Color(0xFFC62828), icon: Icons.error);
      case BannerType.info:
        return (bg: const Color(0xFFE8F0FE), fg: const Color(0xFF1A5DC7), icon: Icons.info);
    }
  }

  @override
  Widget build(BuildContext context) {
    final s = _style();
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 14),
      decoration: BoxDecoration(
        color: s.bg,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: s.fg.withValues(alpha: 0.25)),
      ),
      child: Row(
        children: [
          Icon(s.icon, color: s.fg, size: 22),
          const SizedBox(width: 12),
          Expanded(
            child: Text(
              message,
              style: TextStyle(color: s.fg, fontWeight: FontWeight.w500),
            ),
          ),
          if (onClose != null)
            InkWell(
              onTap: onClose,
              child: Icon(Icons.close, color: s.fg, size: 18),
            ),
        ],
      ),
    );
  }
}

File: android_native/app/src/main/java/com/example/widgetgallery/presentation/components/AppBanner.kt
package com.example.widgetgallery.presentation.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Error
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.WarningAmber
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

enum class BannerType { Info, Success, Warning, Error }

/** Inline info / success / warning / error banner with dismiss. */
@Composable
fun AppBanner(
    message: String,
    modifier: Modifier = Modifier,
    type: BannerType = BannerType.Info,
    onClose: (() -> Unit)? = null,
) {
    val (bg, fg, icon) = style(type)
    Row(
        modifier
            .clip(RoundedCornerShape(12.dp))
            .background(bg)
            .border(1.dp, fg.copy(alpha = 0.25f), RoundedCornerShape(12.dp))
            .padding(horizontal = 16.dp, vertical = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, null, tint = fg, modifier = Modifier.size(22.dp))
        Spacer(Modifier.width(12.dp))
        Text(message, color = fg, fontWeight = FontWeight.Medium,
            modifier = Modifier.weight(1f))
        if (onClose != null) {
            Spacer(Modifier.width(8.dp))
            Icon(Icons.Filled.Close, null, tint = fg,
                modifier = Modifier.size(18.dp).clickable(onClick = onClose))
        }
    }
}

private data class BannerStyle(val bg: Color, val fg: Color, val icon: ImageVector)

private fun style(t: BannerType): BannerStyle = when (t) {
    BannerType.Success -> BannerStyle(Color(0xFFE6F4EA), Color(0xFF1E7E34),
        Icons.Filled.CheckCircle)
    BannerType.Warning -> BannerStyle(Color(0xFFFFF4E5), Color(0xFFB26A00),
        Icons.Filled.WarningAmber)
    BannerType.Error -> BannerStyle(Color(0xFFFDECEA), Color(0xFFC62828),
        Icons.Filled.Error)
    BannerType.Info -> BannerStyle(Color(0xFFE8F0FE), Color(0xFF1A5DC7),
        Icons.Filled.Info)
}
